### DISCUSSION TITLE
Cannot read properties of undefined (reading '')

### DISCUSSION THREAD

**Discussion by dhduvall** [renovatebot/renovate#22953] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.140.3

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm messing with `platform=local`, trying to get renovate to run on a local git repo. I'm getting a message `Repository has unknown error`, with an error message of `Cannot read properties of undefined (reading '')` and the stack
```
TypeError: Cannot read properties of undefined (reading '')
    at getBranchCommit (/Users/duvall/src/renovate/lib/util/git/index.ts:503:30)
    at checkOnboardingBranch (/Users/duvall/src/renovate/lib/workers/repository/onboarding/branch/index.ts:120:20)
    at getRepoConfig (/Users/duvall/src/renovate/lib/workers/repository/init/config.ts:11:12)
    at initRepo (/Users/duvall/src/renovate/lib/workers/repository/init/index.ts:43:12)
    at Object.renovateRepository (/Users/duvall/src/renovate/lib/workers/repository/index.ts:54:14)
    at attributes.repository (/Users/duvall/src/renovate/lib/workers/global/index.ts:184:11)
    at start (/Users/duvall/src/renovate/lib/workers/global/index.ts:169:7)
    at /Users/duvall/src/renovate/lib/renovate.ts:18:22
```
I have a fix, but I'm uncertain about the way I've structured it—I just check `GlobalConfig.get('platform')` in `fetchBranchCommits()`. Maybe a better way is to make a platform method to return the command to get the ref/hash mapping, but I didn't see how I could write all the non-local platform versions once, rather than copy it into each. Could I test for the existence of the method and fall back to the current code if it doesn't exist?

I'm also struggling to figure out how to test this.

I can post a PR if that would help, but wanted to start the discussion first as suggested by the contribution guidelines.

### Relevant debug logs

https://gist.github.com/dhduvall/94b66eae4c535992a3b6929b1a25ec2f

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Is your repo onboarded? Please share the full debug log too

**Reply by dhduvall** [reply]
The repo is not onboarded. I edited the original message to include a pointer to the full debug log.

**Reply by rarkins** [reply]
Onboarding with platform=local is a tough one. Maybe we should bypass all onboarding logic by setting requireConfig=optional. That way it will consider the repo to be "onboarded" and not try to create an onboarding branch

**Reply by dhduvall** [reply]
I guess it depends on what the purpose of `platform=local` is. I've been thinking of it as a way to test `renovate` functionality without having to—in the target repo—create commits, push them, etc.

Maybe a better word is "evaluate"?

Assuming the (a?) goal is this testing/evaluation, the question remains what of `renovate`'s functionality is on the table for that. It might be nice to test the onboarding functionality, too, though I'll admit that wasn't my aim here, specifically, and when I say I "got it to work", it did in fact not create an onboarding branch, it just told me what it wanted to upgrade—i.e., it gave me the `packageFiles with updates` message, with all of the juicy details.

That said, `requireConfig=optional` gets me to the same place.

**Comment by dhduvall** [comment]
Actually, I realized I don't need the changes I made to `fetchBranchCommits()`, only the addition of `git.initRepo()` in the local platform's `initRepo()`.

**Comment by pat-s** [comment]
I was facing the same error and the following worked for me

```
RENOVATE_ONBOARDING=false LOG_LEVEL=debug renovate  --platform=local
```

`RENOVATE_REQUIRE_CONFIG=ignored` had no effect.

**Comment by juliancarrivick** [comment]
In case this helps anyone else I also was facing this error, disabling onboarding let me run renovate but nothing was happening. Once I enabled debug logging I could see it was checking Git for the file, not the local file system. I hadn't checked in my renovate config into Git yet so it wasn't finding it. Once I did that it ran successfully.

**Reply by viceice** [reply]
@rarkins it seems we should update this docs to mention it needs a repo config?

https://docs.renovatebot.com/modules/platform/local/

**Reply by rarkins** [reply]
Docs update would be good, but maybe it should automatically run with requireConfig=optional?
